x-pack/legacy/plugins/siem/public/containers/case/use_get_cases.tsx

```diff
@@ -4,136 +4,185 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { Dispatch, SetStateAction, useEffect, useReducer, useState } from 'react';
+import { Dispatch, SetStateAction, useCallback, useEffect, useReducer, useState } from 'react';
 import { isEqual } from 'lodash/fp';
-import {
-  DEFAULT_TABLE_ACTIVE_PAGE,
-  DEFAULT_TABLE_LIMIT,
-  FETCH_FAILURE,
-  FETCH_INIT,
-  FETCH_SUCCESS,
-  UPDATE_QUERY_PARAMS,
-  UPDATE_FILTER_OPTIONS,
-} from './constants';
-import { AllCases, SortFieldCase, FilterOptions, QueryParams } from './types';
-import { getTypedPayload } from './utils';
+import { DEFAULT_TABLE_ACTIVE_PAGE, DEFAULT_TABLE_LIMIT } from './constants';
+import { AllCases, SortFieldCase, FilterOptions, QueryParams, Case } from './types';
 import { errorToToaster } from '../../components/ml/api/error_to_toaster';
 import { useStateToaster } from '../../components/toasters';
 import * as i18n from './translations';
-import { getCases } from './api';
+import { UpdateByKey } from './use_update_case';
+import { getCases, updateCaseProperty } from './api';
 
 export interface UseGetCasesState {
+  caseCount: CaseCount;
   data: AllCases;
-  isLoading: boolean;
+  filterOptions: FilterOptions;
   isError: boolean;
+  isLoading: boolean;
```

Comment:
do we really need `isLoading` and `loading`?

Reply:
this loads different things, `cases`, `caseCount`, and updates to 'case.state'

Reply:
i had it explicitly typed to take cases and caseCount but then case.state messed up my game, i can go back to that? but i bet you have other suggestions ;)

Reply:
ok we will refactor when the API is smarter